pkgs/applications/video/shotcut/default.nix

```diff
@@ -14,65 +16,59 @@
 , qtwebsockets
 , qtquickcontrols2
 , qtgraphicaleffects
-, qmake
 , qttools
+, qttranslations
 , gitUpdater
+, wrapQtAppsHook
 }:
 
-assert lib.versionAtLeast mlt.version "6.24.0";
+assert lib.versionAtLeast mlt.version "7.6.0";
 
 mkDerivation rec {
   pname = "shotcut";
-  version = "21.09.20";
+  version = "22.10.25";
 
   src = fetchFromGitHub {
     owner = "mltframework";
     repo = "shotcut";
     rev = "v${version}";
-    sha256 = "1y46n5gmlayfl46l0vhg5g5dbbc0sg909mxb68sia0clkaas8xrh";
+    sha256 = "ldlKtOYIZcoRddYYT1YYRyn+EmqV/1w63b9WH1o6wYo=";
   };
 
-  nativeBuildInputs = [ pkg-config qmake ];
+  nativeBuildInputs = [ cmake pkg-config wrapQtAppsHook ];
   buildInputs = [
     SDL2
+    fftw
     frei0r
     ladspaPlugins
     gettext
     mlt
     qtbase
     qtmultimedia
+    qttranslations
     qtx11extras
     qtwebsockets
     qtquickcontrols2
     qtgraphicaleffects
   ];
 
-  NIX_CFLAGS_COMPILE = "-I${mlt.dev}/include/mlt++ -I${mlt.dev}/include/mlt";
-  qmakeFlags = [
-    "QMAKE_LRELEASE=${lib.getDev qttools}/bin/lrelease"
-    "SHOTCUT_VERSION=${version}"
-    "DEFINES+=SHOTCUT_NOUPGRADE"
-  ];
-
   prePatch = ''
-    sed 's_shotcutPath, "melt[^"]*"_"${mlt}/bin/melt"_' -i src/jobs/meltjob.cpp
+    sed 's_shotcutPath, "melt[^"]*"_"${mlt}/bin/melt"_' -i src/jobs/meltjob.cpp src/mltcontroller.cpp
     sed 's_shotcutPath, "ffmpeg"_"${mlt.ffmpeg}/bin/ffmpeg"_' -i src/jobs/ffmpegjob.cpp
-    sed 's_qApp->applicationDirPath(), "ffmpeg"_"${mlt.ffmpeg}/bin/ffmpeg"_' -i src/docks/encodedock.cpp
+    sed 's_qApp->applicationDirPath(), "ffmpeg"_"${mlt.ffmpeg}/bin/ffmpeg"_' -i src/docks/encodedock.cpp src/widgets/directshowvideowidget.cpp
+    sed 's_shotcutPath, "ffprobe"_"${mlt.ffmpeg}/bin/ffprobe"_' -i src/jobs/ffprobejob.cpp
     NICE=$(type -P nice)
     sed "s_/usr/bin/nice_''${NICE}_" -i src/jobs/meltjob.cpp src/jobs/ffmpegjob.cpp
   '';
 
   qtWrapperArgs = [
-    "--prefix FREI0R_PATH : ${frei0r}/lib/frei0r-1"
-    "--prefix LADSPA_PATH : ${ladspaPlugins}/lib/ladspa"
     "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ jack1 SDL2 ]}"
-    "--prefix PATH : ${mlt}/bin"
```

Comment:
I've removed the three lines above, as I haven't noticed they're being needed for anything, but maybe it's just functionality I didn't use and they should stay?

Reply:
`FREI0R_PATH` and `LADSPA_PATH` were needed to find some audio filters. Those filters are now missing. Unfortunately re-adding those two lines does not fix this, so this needs investigation.
Setting `PATH` seems not necessary anymore because the path to "melt" gets already patched in.

Reply:
Thanks. I'll try to have a look, can you better describe where the filters are supposed to appear so that I can check for them?

Reply:
`egrep -r 'mlt_service:.*(frei0r|ladspa)' ~/git/shotcut/src/qml/filters/`

Reply:
Did either of you figure out the plugin issue?

I noticed that (on Arch Linux), if I use the `makepkg` (`pacman`) built binary and set `LADSPA_PATH` manually to `/nix/store/....-swh-plugins-0.4.17/lib/ladspa/` the plugins are detected just fine; however, the `nix`-built binary still doesn't (with the abovementioned lines).

Seems like something is weird with the nixpkgs Qt wrapper, but I don't know how to diagnose further